galsim/utilities.py

```diff
@@ -1208,3 +1208,84 @@ def generate():
             c *= b_over_a * (n-i)/(i+1)
             yield c
     return np.fromiter(generate(), float, n+1)
+
+def rand_with_replacement(n, rng, n_choices, weight=None, _n_rng_calls=False):
```

Comment:
Looks good.  My only thought is that we might want to reorder the call signature to `n, n_choices, rng`, so that once #824 is merged and `None` becomes a valid seed for `UniformDeviate` we can transparently make `rng=None` the default here.